rules/doctrine-code-quality/src/Rector/Property/RemoveRedundantDefaultAnnotationValuesRector.php

```diff
@@ -0,0 +1,250 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DoctrineCodeQuality\Rector\Property;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\Property;
+use Rector\BetterPhpDocParser\ValueObject\PhpDocNode\Doctrine\AbstractDoctrineTagValueNode;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Rector\DoctrineCodeQuality\NodeAnalyzer\DoctrineClassAnalyzer;
+use Rector\DoctrineCodeQuality\NodeAnalyzer\DoctrinePropertyAnalyzer;
+
+/**
+ * @see \Rector\DoctrineCodeQuality\Tests\Rector\Property\RemoveRedundantDefaultAnnotationValuesRector\RemoveRedundantDefaultAnnotationValuesRectorTest
+ * @see \Rector\DoctrineCodeQuality\Tests\Rector\Class_\RemoveRedundantDefaultAnnotationValuesRector\RemoveRedundantDefaultAnnotationValuesRectorTest
+ */
+final class RemoveRedundantDefaultAnnotationValuesRector extends AbstractRector
+{
+    /**
+     * @var DoctrinePropertyAnalyzer
+     */
+    private $doctrinePropertyAnalyzer;
+
+    /**
+     * @var DoctrineClassAnalyzer
+     */
+    private $doctrineClassAnalyzer;
+
+    /**
+     * @var bool
+     */
+    private $hasModifiedAnnotation = false;
+
+    public function __construct(
+        DoctrinePropertyAnalyzer $doctrinePropertyAnalyzer,
+        DoctrineClassAnalyzer $doctrineClassAnalyzer
+    ) {
+        $this->doctrinePropertyAnalyzer = $doctrinePropertyAnalyzer;
+        $this->doctrineClassAnalyzer = $doctrineClassAnalyzer;
+
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition(
+            'Removes redundant default values from Doctrine ORM annotations.',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity
+ */
+class SomeClass
+{
+    /**
+     * @ORM\ManyToOne(targetEntity=Training::class)
+     * @ORM\JoinColumn(name="training", unique=false)
+     */
+    private $training;
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity
+ */
+class SomeClass
+{
+    /**
+     * @ORM\ManyToOne(targetEntity=Training::class)
+     * @ORM\JoinColumn(name="training")
+     */
+    private $training;
+}
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class, Property::class];
```

Comment:
It would be great to have these 2 separated Rector rules. The class is huge now and t would could me problematic to debug it later.

One for `Class_`, one for `Property`. It also answers your question above, now I get it :)

Reply:
Done 👍